### DISCUSSION TITLE
NvChad in the Lazyman Neovim Configuration Manager

### DISCUSSION THREAD

**Discussion by doctorfree** [NvChad/NvChad#2013] [General]
I've created a Neovim Configuration Manager that supports easy installation and initialization of multiple Neovim configurations using the `NVIM_APPNAME` environment variable in Neovim 0.9.  I call it Lazyman and include support for `NvChad` along with my own NvChad customization at https://github.com/doctorfree/NvChad-custom based on the NvChad example.

See https://github.com/doctorfree/nvim-lazyman for info on Lazyman.

I'm contacting the maintainers of the Lazyman supported Neovim configurations to seek any feedback on the description, installation, and initialization of their config in Lazyman. Please feel free to let me know what I'm doing wrong or how I could improve the incorporation of `NvChad` in Lazyman. If you feel there is a need to modify the way `NvChad` is used in Lazyman then open an issue at https://github.com/doctorfree/nvim-lazyman/issues

NvChad is very cool. Thanks!

**Comment by siduck** [comment]
hi @doctorfree good work! but please avoid using external themes in nvchad, they'd slow down it and defeat purpose of our performant base46 plugin :(

I did port tokyonight to base46

**Reply by doctorfree** [reply]
thanks @siduck i was unaware of the features available with the base46 plugin and the default theme. it's amazing! this and the cheatsheet are two distinguishing features of NvChad.

i've revised my NvChad custom repo to take advantage of this as well as some keymaps to make it easy to toggle theme and transparency. i think the area most in need of customization is lsp and mason. i was also unaware of the extras in the `v2.0_featureful` branch. that's a good start, now i need to learn how to integrate my terminal configuration into Nvterm.

thanks for the feedback, i learned a lot pretty quickly and easily.

**Reply by siduck** [reply]
btw there are other features too, check https://nvchad.com/docs/features

**Reply by doctorfree** [reply]
I will explore these. Such an excellent Neovim configuration distribution!

One of the things I have done with the Lazyman Neovim default configuration is to provide a terminal user interface with a menu system to enable, disable, and configure settings in the configuration. This interface is then embedded in the Neovim terminal plugin I use, Tommaso Laurenzi's  [terminal.nvim](https://github.com/rebelot/terminal.nvim). The user can, with a simple keymap, bring up a menu to toggle or set various configuration parameters.

All of this is an attempt to make it easier for folks to get started and utilize many of the advanced features available in these configs.

Now that I've played with it a little more, I really like NvChad. I will put more effort into tailoring my custom NvChad config to take advantage of these excellent features and I plan to add a section to the Lazyman wiki describing in greater detail what each supported config provides.

**Comment by doctorfree** [comment]
I added the basic Neovim configuration at https://github.com/NvChad/basic-config to the "Starter" Lazyman configurations. It's an informative config for the new user and works well without modification with the `lazyman` command. Install and initialize the `Basic` Lazyman starter configuration with the command `lazyman -x Basic`.

**Comment by doctorfree** [comment]
I've improved the [custom NvChad configuration](https://github.com/doctorfree/NvChad-custom) used in the `lazyman` installation and initialization of `NvChad`. I'm pretty happy with it but would welcome any further suggestions or criticism. The lazyman customizations of NvChad are primarily around LSP, Mason, and `lspconfig`, diagnostics, and auto-install of Mason packages during `lazyman` initialization.

I'm marking this discussion as having been "answered", the feedback above sufficiently inspiring me to make these improvements. Thanks!
